discretisedfield/field.py

```diff
@@ -3059,112 +3061,98 @@ def _fromovf(cls, filename):
-            with open(filename, 'r', encoding='utf-8') as ovffile:
-                lines = ovffile.readlines()
-
-            mdatalines = list(filter(lambda s: s.startswith('#'), lines))
-            datalines = np.genfromtxt(filter(lambda s: not s.startswith('#'),
-                                             lines), dtype=None)
-
-            if '1.0' in mdatalines[0]:
-                # valuedim is not in OVF1 file.
-                mdatadict['valuedim'] = datalines.shape[-1]
-
-            for line in mdatalines:
-                for mdatum in mdatalist:
-                    if mdatum in line:
-                        mdatadict[mdatum] = float(line.split()[-1])
-                        break
-
-        except UnicodeDecodeError:
-            with open(filename, 'rb') as ovffile:
-                f = ovffile.read()
-                lines = f.split(b'\n')
-
-            mdatalines = list(filter(lambda s: s.startswith(bytes('#',
-                                                                  'utf-8')),
-                                     lines))
-
-            if bytes('2.0', 'utf-8') in mdatalines[0]:
-                endian = '<'  # little-endian
-            elif bytes('1.0', 'utf-8') in mdatalines[0]:
-                endian = '>'  # big-endian
-
-            for line in mdatalines:
-                for mdatum in mdatalist:
-                    if bytes(mdatum, 'utf-8') in line:
-                        mdatadict[mdatum] = float(line.split()[-1])
-                        break
-
-            header = b'# Begin: Data Binary '
-            data_start = f.find(header)
-            header = f[data_start:(data_start + len(header) + 1)]
-
-            data_start += len(header)
-            data_end = f.find(b'# End: Data Binary ')
-
-            if b'4' in header:
-                nbytes = 4
-                formatstr = endian + 'f'
-                checkvalue = 1234567.0
-            elif b'8' in header:
-                nbytes = 8
-                formatstr = endian + 'd'
-                checkvalue = 123456789012345.0
-
-            newlines = [b'\n\r', b'\r\n', b'\n']  # ordered by length
-            for nl in newlines:
-                if f.startswith(nl, data_start):
-                    data_start += len(nl)
-                    # There is a difference between files written by OOMMF and
-                    # mumax3. OOMMF has a newline character before the "end
-                    # metadata line', whereas mumax3 does not. Therefore if the
-                    # length of the data stream is not a multiple of nbytes, we
-                    # have to subtract the length of newline character from
-                    # data_end.
-                    if (data_end - data_start) % nbytes != 0:
-                        data_end -= len(nl)
+        mdata_str = ['meshunit', 'valueunit']  # information not used
+        mdata_list = ['valuelabels', 'valueunits']  # information not used
+
+        header = {}
```

Comment:
I do not know if we can remove `mdata_float`, `mdata_int`, etc. and just use `header = {}`. This way we add all metadata information to `header`. Later, we can use `header.get('xmin', None)` when we need it. This might simplify the code and reduce line count.

Reply:
Also, we read all metadata to a dictionary and we can reuse it in the future more easily in the future.

Reply:
The different lists are mainly used to do the correct conversion (`float`, `int`, ...) right away. Somewhere we need to store the information of what to do with which value and with the current implementation all of this is in one place.

Reply:
I've moved it now.